Packages/com.chisel.components/Chisel/Components/API.public/BrushMeshAssetFactory/SpiralStairs/CSGSpiralStairsDefinition.cs

```diff
@@ -5,143 +5,178 @@
-
-			riserType		= StairsRiserType.ThickRiser;
-			riserDepth		= kDefaultRiserDepth;
-
-			bottomSmoothingGroup = 0;
-		}
-
-		public void Validate()
-		{
-			stepHeight		= Mathf.Max(kMinStepHeight, stepHeight);
-			
-			innerDiameter	= Mathf.Min(outerDiameter - kMinStairsDepth,  innerDiameter);
-			innerDiameter	= Mathf.Max(kMinInnerDiameter,  innerDiameter);
-			outerDiameter	= Mathf.Max(innerDiameter + kMinStairsDepth,  outerDiameter);
-			outerDiameter	= Mathf.Max(kMinOuterDiameter,  outerDiameter);
-			height			= Mathf.Max(stepHeight, height);
-			treadHeight		= Mathf.Max(0, treadHeight);
-			nosingDepth		= Mathf.Max(0, nosingDepth);
-			nosingWidth		= Mathf.Max(0, nosingWidth);
-
-			riserDepth		= Mathf.Max(kMinRiserDepth, riserDepth);
-
-			rotation		= Mathf.Max(kMinRotation, Mathf.Abs(rotation)) * (rotation < 0 ? -1 : 1);
-
-			innerSegments	= Mathf.Max(kMinSegments, innerSegments);
-			outerSegments	= Mathf.Max(kMinSegments, outerSegments);
-		}
-	}
+    {
+        public const float	kMinStepHeight			= 0.01f;
+        public const float  kMinStairsDepth         = 0.1f;
+        public const float  kMinRiserDepth          = 0.01f;
+        public const float	kMinRotation			= 15;
+        public const int    kMinSegments            = 3;
+        public const float	kMinInnerDiameter		= 0.00f;
+        public const float	kMinOuterDiameter		= 0.01f;
+        
+        public const float	kDefaultStepHeight      = 0.20f;
+        public const float	kDefaultTreadHeight     = 0.02f;
+        public const float	kDefaultNosingDepth     = 0.02f;
+        public const float	kDefaultNosingWidth     = 0.01f;
+
+        public const float	kDefaultInnerDiameter	= 0.25f;
+        public const float	kDefaultOuterDiameter	= 2;
+        public const float	kDefaultHeight          = 1;
+        
+        public const int	kDefaultInnerSegments	= 8;
+        public const int	kDefaultOuterSegments	= 16;
+        
+        public const float	kDefaultStartAngle		= 0;
+        public const float	kDefaultRotation		= 180;
+
+        public const float  kDefaultRiserDepth      = 0.03f;
+        
+        [DistanceValue] public Vector3  origin;
+        [DistanceValue] public float	height;
+        [DistanceValue] public float	outerDiameter;
+        [DistanceValue] public float	innerDiameter;
+        [DistanceValue] public float	stepHeight;
+        [DistanceValue] public float	treadHeight;
+        [DistanceValue] public float	nosingDepth;
+        [DistanceValue] public float	nosingWidth;
+        [DistanceValue] public float	riserDepth;
+        [AngleValue   ] public float	startAngle;
+        [AngleValue   ] public float	rotation; // can be >360 degrees
+        public int						innerSegments;
+        public int						outerSegments;
+        public StairsRiserType			riserType;
+
+        public uint						bottomSmoothingGroup;
+
+        public CSGSurfaceAsset[]        surfaceAssets;
+        public SurfaceDescription[]     surfaceDescriptions;
```

Comment:
Moved the surface assets/descriptions out of the component and into the definition where it belongs